File: source/cqut-online/src/test/java/com/chronos/mobile/source/cqutonline/CqutCasPasswordEncryptorTest.kt
package com.chronos.mobile.source.cqutonline

import org.junit.Assert.assertEquals
import org.junit.Assert.assertTrue
import org.junit.Test

class CqutCasPasswordEncryptorTest {
    private val encryptor = CqutCasPasswordEncryptor()

    @Test
    fun `encrypt returns empty string for blank password`() {
        assertEquals("", encryptor.encrypt(""))
    }

    @Test
    fun `encrypt returns encoded payload for non blank password`() {
        val encrypted = encryptor.encrypt("123456")
        assertTrue(encrypted.isNotBlank())
        assertTrue(encrypted.contains("%") || encrypted.contains("%5B"))
    }
}

File: src/lib/parsers/cqut-online/cqut-cas-password-encryptor.test.ts
import { describe, expect, it } from 'vite-plus/test';
import { encryptCasPassword } from '$lib/client/cqut-cas-password-encryptor';

describe('encryptCasPassword', () => {
	it('returns empty string for blank password', async () => {
		await expect(encryptCasPassword('')).resolves.toBe('');
		await expect(encryptCasPassword('   ')).resolves.toBe('');
	});

	it('returns encoded payload for non blank password', async () => {
		const encrypted = await encryptCasPassword('123456');
		expect(encrypted.length).toBeGreaterThan(0);
		expect(encrypted.includes('%') || encrypted.includes('%5B')).toBe(true);
	});

	it('splits long passwords into 30-char encrypted chunks', async () => {
		const singleChunk = JSON.parse(
			decodeURIComponent(await encryptCasPassword('a'.repeat(30)))
		) as string[];
		expect(singleChunk).toHaveLength(1);

		const twoChunks = JSON.parse(
			decodeURIComponent(await encryptCasPassword('a'.repeat(60)))
		) as string[];
		expect(twoChunks).toHaveLength(2);

		const threeChunks = JSON.parse(
			decodeURIComponent(await encryptCasPassword('a'.repeat(61)))
		) as string[];
		expect(threeChunks).toHaveLength(3);
	});
});
